src/main/java/net/gegy1000/plasmid/block/CustomBlock.java

```diff
@@ -0,0 +1,97 @@
+package net.gegy1000.plasmid.block;
+
+import com.google.common.base.Preconditions;
+import it.unimi.dsi.fastutil.longs.Long2ObjectArrayMap;
+import net.gegy1000.plasmid.registry.TinyRegistry;
+import net.minecraft.block.BlockState;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.Set;
+
+public class CustomBlock {
+    private static final TinyRegistry<CustomBlock> REGISTRY = TinyRegistry.newStable();
+    private static final Long2ObjectArrayMap<CustomBlock> BLOCK_MAP = new Long2ObjectArrayMap<>();
```

Comment:
This probably works for now, but I guess there's the worry that the block is updated in the world but the custom block isn't. Mostly notably being chunks being totally removed when games close

Reply:
i might work on that in a future pr. i doubt anyone will run 2 games at once anyways. it crashes somewhere else if you do